sale_import_base/models/sale_channel_importer.py

```diff
@@ -0,0 +1,295 @@
+        vals = self._prepare_partner(customer_data)
+        if partner:
+            partner.write(vals)
+            return partner
+        else:
+            partner = self.env["res.partner"].create(vals)
+            self._binding_partner(partner, customer_data["external_id"])
+            return partner
+
+    def _find_partner(self, customer_data):
+        channel = self.payload_id.sale_channel_id
+        external_id = customer_data["external_id"]
+        binding = self.env["sale.channel.partner"].search(
+            [
+                ("external_id", "=", external_id),
+                ("sale_channel_id", "=", channel.id),
+            ]
+        )
+        if binding:
+            return binding.partner_id
+        elif channel.allow_match_on_email:
+            partner = self.env["res.partner"].search(
+                [("email", "=", customer_data["email"])]
+            )
+            if partner:
+                self._binding_partner(partner, customer_data["external_id"])
+                return partner
+
+    def _prepare_partner(self, data, parent_id=None, archived=None):
+        result = {
+            "name": data["name"],
+            "street": data.get("street"),
+            "street2": data.get("street2"),
+            "zip": data.get("zip"),
+            "city": data.get("city"),
+            "email": data.get("email"),
+            "phone": data.get("phone"),
+            "mobile": data.get("mobile"),
+        }
+        if parent_id:
+            result["parent_id"] = parent_id
+        if archived:
+            result["active"] = False
+        if data.get("country_code"):
+            country = self.env["res.country"].search(
+                [("code", "=", data["country_code"])]
+            )
+            if not country:
+                raise ValidationError(
+                    _("Missing country {}").format(data["country_code"])
+                )
+            result["country_id"] = country.id
+            if data.get("state_code"):
+                state = self.env["res.country.state"].search(
+                    [("code", "=", data["state_code"]), ("country_id", "=", country.id)]
+                )
+                if not state:
+                    raise ValidationError(
+                        _("Missing State %(state_code)s for country %(country_name)s")
+                        % {
+                            "state_code": data["state_code"],
+                            "country_name": country.name,
+                        }
+                    )
+                result["state_id"] = state.id
+        return result
+
+    def _should_merge_addresses(self, vals_addr_invoice, vals_addr_shipping):
+        return vals_addr_invoice == vals_addr_shipping
+
+    def _process_addresses(self, parent, address_invoice, address_shipping):
+        vals_addr_invoice = self._prepare_partner(address_invoice, parent.id, True)
+        vals_addr_shipping = self._prepare_partner(address_shipping, parent.id, True)
```

Comment:
@florian-dacosta,

@nayatec suggested in a recent PR that I copied [in this last commit](https://github.com/akretion/sale-import/pull/95/commits/71e76d095d11cd25d1f09250331ba430868d8fdf), to add a `archive_addresses` boolean field instead of this always `True` parameter. Allowing the user to choose for each channel if he wants to archive the invoice/shipping addresses or not.

Do you think it's a good idea to implement in this sale_import_base module?

Reply:
Yes, definitly !

Reply:
It does not exist yet! :)
But I think you might be able to cherry pick this last one...

Reply:
Ok, I mis-understood, I'll cherry-pick it